### ISSUE TITLE
Bbox normalization

### ISSUE THREAD

**Issue by AldusLorde** [fizyr/keras-retinanet#1273]
![Screenshot from 2020-02-11 09-14-59](https://user-images.githubusercontent.com/34024120/74217046-ffed8c80-4cae-11ea-9645-a81ab66c1d5c.png)

Hi! What are these magic constants? It's written in the Fast R-CNN paper that the authors normalize bboxes. According to their code, they calculate the mean and variance dynamically and don't use precalculated values. On the other hand, Faster R-CNN uses the same values as you...
Are these value precalculated for Pascal VOC / MS COCO?

According to the paper, RetinaNet uses the standard R-CNN anchor parametrization (tx, ty, th, tw).
On the other hand, you use a custom parametrization:
![Screenshot from 2020-02-11 09-39-24](https://user-images.githubusercontent.com/34024120/74218277-69bb6580-4cb2-11ea-8c84-53ee8cc846b1.png)

What is a purpose of doing so?

**Comment by hgaiser** [comment]
> Hi! What are these magic constants?

They are computed from COCO. The mean is roughly 0 and the std is roughly 0.2. Instead of taking their exact values, they are rounded to the nearest (nice) values.

> According to the paper, RetinaNet uses the standard R-CNN anchor parametrization (tx, ty, th, tw).
> On the other hand, you use a custom parametrization:

The rest of the code uses notations `[x1, y1, x2, y2]`, but the parametrization used `[x_ctr, y_ctr, h, w]` before. I ran a test, changing it to the current parametrization and got marginally better results. I don't think it performs better, but it at least doesn't perform worse ;)

**Comment by AldusLorde** [comment]
> > Hi! What are these magic constants?
> 
> They are computed from COCO. The mean is roughly 0 and the std is roughly 0.2. Instead of taking their exact values, they are rounded to the nearest (nice) values.
> 
> > According to the paper, RetinaNet uses the standard R-CNN anchor parametrization (tx, ty, th, tw).
> > On the other hand, you use a custom parametrization:
> 
> The rest of the code uses notations `[x1, y1, x2, y2]`, but the parametrization used `[x_ctr, y_ctr, h, w]` before. I ran a test, changing it to the current parametrization and got marginally better results. I don't think it performs better, but it at least doesn't perform worse ;)

That's interesting, thank you. Could you please add some documentation about these things somewhere?

**Comment by hgaiser** [comment]
@AldusLorde could you do it and make a PR for that?

**Comment by HyunjiEllenPak** [comment]
@hgaiser 
Hi, I know the mean and std are used to normalize the ground-truth regression targets to have zero mean and unit variance. And the default values (mean=0, std=0.2) were calculated from coco-dataset. Then, If I'd like to train on custom datasets, how could I calculate the mean and std values?
